domainservice/src/main/kotlin/systems/ajax/malov/stockanalyzer/repository/StockRecordRepository.kt

```diff
@@ -8,11 +8,11 @@ import java.util.Date
 interface StockRecordRepository {
     fun insertAll(mongoStockRecords: List<MongoStockRecord>): Flux<MongoStockRecord>
 
-    fun findTopNStockSymbolsWithStockRecords(
-        n: Int,
+    fun findTopStockSymbolsWithStockRecords(
+        quantity: Int,
         from: Date,
         to: Date,
-    ): Mono<Map<String, List<MongoStockRecord>>>
+    ): Mono<LinkedHashMap<String, List<MongoStockRecord>>>
```

Comment:
```suggestion
    ): Mono<Map<String, List<MongoStockRecord>>>
```

Or create a custom object that aggregates the result.
Don't rely on concrete implementation